src/windows/osd.c

```diff
@@ -207,33 +207,36 @@ int ofi_shm_map(struct util_shm *shm, const char *name, size_t size,
 		if (!shm->shared_fd) {
 			FI_WARN(&core_prov, FI_LOG_CORE, "CreateFileMapping failed\n");
 			ret = -FI_EINVAL;
-			goto fn_fail;
+			goto fn_nofilemap;
 		}
 	} else { /* readonly */
 		shm->shared_fd = OpenFileMappingA(access, FALSE, shm->name);
 		if (!shm->shared_fd) {
 			FI_WARN(&core_prov, FI_LOG_CORE, "OpenFileMapping failed\n");
 			ret = -FI_EINVAL;
-			goto fn_fail;
+			goto fn_nofilemap;
 		}
 	}
 
 	shm->ptr = MapViewOfFile(shm->shared_fd, access, 0, 0, size);
 	if (!shm->ptr) {
 		FI_WARN(&core_prov, FI_LOG_CORE, "MapViewOfFile failed\n");
 		ret = -FI_EINVAL;
-		goto fn_fail;
+		goto fn_nomap;
 	}
 
+	/* size value not really used due to missing remap functionality,
+	   but may be useful for debugging */
+	shm->size = size;
 	*mapped = shm->ptr;
 
 	return FI_SUCCESS;
 
-fn_fail:
-	if (shm->shared_fd)
-		CloseHandle(shm->shared_fd);
-	if (fname)
-		free((void*)fname);
+fn_nomap:
+	CloseHandle(shm->shared_fd);
+fn_nofilemap:
+	free((void*)fname);
```

Comment:
FWIW: I'm all for paranoid compiler configurations, but one that forces excessive unnecessary casts is not a good configuration.